Add angle method to calculate angle between two Vectors (issue #31)

josephjcontreras opened the pull request:
The angle method will calculate the angle between two
Vectors (both Vector2 or Vector3) and return the angle
either radians (default) or degrees

This is in response to issue #31.

fwkoch commented on vectormath/vector.py:
```diff
@@ -85,6 +85,21 @@ def cross(self, vec):
+        if unit == 'rad':
+            return ang
+        else:
+            return ang * 180 / np.pi
```
Looking at travis - there are a couple minor pylint issues. First is this unnecessary `else`/`return`; it can just be:
```python
        if unit == 'rad':
            return ang
        return ang * 180 / np.pi
```
Or possibly even cleaner with one return:
```python
        if unit == 'deg':
            ang = ang * 180 / np.pi
        return ang
```

JJCoding01 replied:
I went with the second option, `if unit == 'deg': ...`

fwkoch commented on vectormath/vector.py:
```diff
@@ -289,6 +304,10 @@ def dot(self, vec):
                              'number of elements.')
         return np.sum((getattr(self, d)*getattr(vec, d) for d in self.dims), 1)
 
+    def angle(self, vec=None):
```
Also from pylint, this call signature should just be the same as the other `angle` method.

fwkoch commented on vectormath/vector.py:
```diff
@@ -85,6 +85,21 @@ def cross(self, vec):
         return self.__class__(np.cross(self, vec))
 
+    def angle(self, vec, unit='rad'):
+        """Angle between two vectors given in either radians ('rad') or degrees
```
Minor, but can you follow this for docstring convention: https://www.python.org/dev/peps/pep-0257/

Basically just one short descriptive line ("Calculate angle between two vectors"), line break, then details (i.e. info about the unit kwarg).

JJCoding01 replied:
Updated docstring to read


```
 """Calculate the angle between two Vectors

unit: unit for returned angle, either 'rad' or 'deg'. Defaults to 'rad'
"""
```

fwkoch commented on vectormath/vector.py:
```diff
@@ -85,6 +85,21 @@ def cross(self, vec):
+        ('deg'). Default to radians
+        """
+        if not isinstance(vec, self.__class__):
+            raise TypeError('Angle operand must be a vector')
```
Another picky comment but possibly this should be more specific (since you can't do angle between Vector2 and 3). Just like `'Angle operand must be a vector of class {}'.format(self.__class__.__name__)`? If that makes sense to you...

JJCoding01 replied:
Good idea. Updated error message to read `'Angle operand must be of class {}'                        .format(self.__class__.__name__)`, like you suggested.

fwkoch commented on vectormath/vector.py:
```diff
@@ -85,6 +85,21 @@ def cross(self, vec):
+        if unit not in ['deg', 'rad']:
+            raise ValueError('Only units of rad or deg are supported')
+
+        ang = np.arccos(self.dot(vec) / (self.length * vec.length))
```
👍

josephjcontreras commented:
@fwkoch Thanks for the feedback. I've updated the `angle` method per your comments to improve the docstring, linting, and error messages.

I also added the `ZeroDivisionError` you suggested for when one of the Vectors has a length of `0`.

Thanks,
